src/System.Net.WebClient/tests/WebClientTest.cs

```diff
@@ -568,6 +577,7 @@ public abstract class WebClientTestBase
                         "\r\n" +
                         $"{largeText}");
                 Assert.Equal(largeText, Encoding.ASCII.GetString(await download));
+                Assert.True(!IsAsync || await downloadProgressInvokedWithContentLength.Task, "Expected download progress callback to be invoked with Content-Length");
```

Comment:
downloadProgressInvokedWithContentLength is never completed with a false result, so the only two outcomes here are either the await will always return true, or the await will hang (it'll never return false and cause the assert to fail).  As such I don't think it should be an assert, instead using an if, and also ideally with a timeout to avoid a hang and instead throw if it doesn't complete in an expected period of time (https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs#L12), e.g.
```C#
if (IsAsync)
{
    await downloadProgressInvokedWithContentLength.TimeoutAfter(30_000);
}
```

Reply:
Good point. Should I update DownloadData_Success and UploadData_Success to use timeouts too or just this test?

Reply:
> Should I update DownloadData_Success and UploadData_Success to use timeouts too or just this test?

Feel free, but not required for this PR; I just want to avoid adding more such issues.